File: src/amtsub/sources/cert_spotter.py
import requests

class CertSpotter:
	async def search(self, domain, timeout):
		try:
			url = f'https://api.certspotter.com/v1/issuances?domain={domain}&include_subdomains=true&expand=dns_names'

			response = requests.get(url, timeout = timeout)

			certificates = response.json()

			subdomains = []

			for certificate in certificates:
				dns_names = certificate['dns_names']

				for subdomain in dns_names:
					subdomains.append(subdomain)

			return subdomains
		except:
			pass

	def get_name(self):
		return 'Cert Spotter'

File: src/amt/sub/sources/cert_spotter/cert_spotter.go
package cert_spotter

import (
	"time"
	"net/http"
)

import (
	jsoniter "github.com/json-iterator/go"
)

type CertSpotter struct {}

type certificateObj struct {
	DNSNames []string `json:"dns_names"`
}

func (self CertSpotter) Search(domain string, timeOut time.Duration) ([]string, error) {
	client := http.Client {
		Timeout: timeOut,
	}

	url := "https://api.certspotter.com/v1/issuances?domain=" + domain + "&include_subdomains=true&expand=dns_names"

	response, err := client.Get(url)

	if err != nil {
		return []string{}, err
	}

	defer response.Body.Close()

	var json = jsoniter.ConfigCompatibleWithStandardLibrary

	decoder := json.NewDecoder(response.Body)

	var certificates []certificateObj

	err = decoder.Decode(&certificates)

	if err != nil {
		return []string{}, err
	}

	var subdomains []string

	for _, certificate := range certificates {
		subdomains = append(subdomains, certificate.DNSNames...)
	}

	return subdomains, nil
}

func (self CertSpotter) GetName() string {
	return "Cert Spotter"
}